airbyte-integrations/connectors/destination-duckdb/destination_duckdb/destination.py

```diff
@@ -109,53 +110,56 @@ def write(
 
             con.execute(query)
 
-        buffer = defaultdict(list)
+        buffer = defaultdict(lambda: defaultdict(list))
 
         for message in input_messages:
             if message.type == Type.STATE:
                 # flush the buffer
                 for stream_name in buffer.keys():
                     logger.info(f"flushing buffer for state: {message}")
-                    table_name = f"_airbyte_raw_{stream_name}"
-                    query = f"""
-                    INSERT INTO {schema_name}.{table_name}
-                      (_airbyte_ab_id, _airbyte_emitted_at, _airbyte_data)
-                    VALUES (?,?,?)
-                    """
-                    con.executemany(query, buffer[stream_name])
+                    DestinationDuckdb._safe_write(con, buffer, schema_name, stream_name)
 
-                con.commit()
```

Comment:
Removed because i think in the existing code this is moot. There was never a transaction to begin with. We also don't need it for the happy path which does the insert in one statement. 
The fallback/execute many path would ideally be wrapped in a transaction, but that is currently not supported by MotherDuck, and  iwould prefer to leave it in its current state.